django/db/models/base.py

```diff
@@ -546,7 +546,10 @@ def __reduce__(self):
 
     def __getstate__(self):
         """Hook to allow choosing the attributes to pickle."""
-        return self.__dict__
+        state = self.__dict__.copy()
+        state['_state'] = copy.copy(state['_state'])
+        state['_state'].fields_cache = {}
```

Comment:
I think we should copy `fields_cache`:
```suggestion
        state['_state'].fields_cache = state['_state'].fields_cache.copy()
```
because a copy of model instance can use the same values and we don't need to fetch it again.

Reply:
Heh, I was gonna try that next ;P Thanks for the suggestions so far felixxm, most have been incorporated.

Django tests pass, I'm still going to test this approach on our code bases.

Reply:
Thanks, the current approach LGTM. Please uncheck "Patch needs improvement" flag on the ticket after local testing.

Reply:
Our test suites passed once I backported this diff to 2.2